core/schema/jsonvalue.go

```diff
@@ -0,0 +1,161 @@
+package schema
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dagger/dagger/core"
+	"github.com/dagger/dagger/dagql"
+)
+
+type jsonvalueSchema struct {
+	srv *dagql.Server
+}
+
+var _ SchemaResolvers = &jsonvalueSchema{}
+
+func (s jsonvalueSchema) Install() {
+	// Top-level constructor: jsonvalue (creates new empty jsonvalue)
+	dagql.Fields[*core.Query]{
+		dagql.Func("json", s.newJSONValue).
+			Doc("Initialize an empty JSON value"),
+	}.Install(s.srv)
+
+	// Expose methods for JSONValue manipulation
+	dagql.Fields[*core.JSONValue]{
+		dagql.Func("get", s.unset).Doc("Return the JSON-encoded value, or a sub-value at the given path").Args(
```

Comment:
Instead of having dedicated `get<Type>`/`set<Type>` methods, what about having just `get`+`set` and then `as<Type>` methods? Then they chain as, `.Get("x").AsInt()`, which feels very much like interfaces...

For `set`, we'd want to have a scalar value that can accept *any* value (I think this is possible, happy to provide pointers, if this seems like a reasonable idea).

Reply:
How would the "universal scalar" type work?